velox/functions/sparksql/tests/ArithmeticTest.cpp

```diff
@@ -417,6 +417,34 @@ TEST_F(ArithmeticTest, isNanDouble) {
   EXPECT_EQ(false, isNan(std::nullopt));
 }
 
+TEST_F(ArithmeticTest, nanvlFloat) {
+  const auto nanvl = [&](std::optional<float> a, std::optional<float> b) {
+    return evaluateOnce<float>("nanvl(c0, c1)", a, b);
+  };
+
+  EXPECT_EQ(0.0f, nanvl(0.0f, kNan));
+  EXPECT_EQ(0.0f, nanvl(kNan, 0.0f));
+  EXPECT_EQ(0.0f, nanvl(0.0f / 0.0f, 0.0f));
+  EXPECT_EQ(std::nullopt, nanvl(0.0f / 0.0f, std::nullopt));
```

Comment:
Can we add test for the first value not being NaN, and the second value being null? That could help verify having nulls in one of the inputs does not imply null in the output.

Reply:
Added, thanks.